### PR TITLE
issue 531: update react native and packages

### PR DESCRIPTION
**Pull request by geosaaga**
- Updated React Native to version 0.71 (was 0.66.5)
- Updated other packages, but tried to avoid breaking changes
- Set iOS deployment target to 12.4
- Updated Readme
- TypeScript fixes because of update packages and newer react version
- Removed old privacy policy link from widget configuration

### CODE REVIEW

File: .node-version [hunk 1]
```diff
@@ -0,0 +1 @@
+18
```
Comment (AnttiRumpunen) [inline .node-version:1]: Olisko node 20 mahdollinen?

File: android/app/src/main/res/layout/widgetconf.xml [hunk 2]
```diff
@@ -180,16 +180,6 @@
...
         android:layout_marginTop="10dip" />
 
-            <TextView
-                android:id="@+id/privacyUrlTextView"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:layout_marginTop="10dip"
-                android:layout_marginBottom="10dip"
-                android:clickable="true"
-                android:linksClickable="true"
```
Comment (AnttiRumpunen) [inline android/app/src/main/res/layout/widgetconf.xml:190]: Joku tekstikenttä pois?

File: .node-version (hunk 1 above)
Comment (geosaaga) [reply .node-version]: 18 tuli react native upgrade helperin kautta, mutta 20 näyttää toimivan myös. Päivitetty .node-version ja Readme

**Review by AnttiRumpunen — APPROVED** [review]
ok

**Pull request merged**